src/query/catalog/src/table.rs

```diff
@@ -247,7 +247,7 @@ pub trait Table: Sync + Send {
         target: CompactTarget,
         limit: Option<usize>,
         pipeline: &mut Pipeline,
-    ) -> Result<Option<Box<dyn TableMutator>>> {
+    ) -> Result<bool> {
```

Comment:
How about adding doc comments for `compact` to explain the meaning of this `bool`?